datum_reader.go

```diff
@@ -185,6 +186,11 @@ func (reader sDatumReader) readValue(field Schema, reflectField reflect.Value, d
 	case Int:
 		return reader.mapPrimitive(func() (interface{}, error) { return dec.ReadInt() })
 	case Long:
+		longSchema := field.(*LongSchema)
+		if longSchema.LogicalType != "" && reflectField.Kind() != reflect.Int64 {
```

Comment:
I just checked and it seems that the method used to set the value requires the target field to be specifically Int64 (no other Int type is assignable).  i.e., it's not using reflect.SetInt so this should be correct.